### PR TITLE
Addresses issue #6519

### PR DESCRIPTION
**Pull request by Garima3110**
Addresses issue #6519

hey @davepagurek , I tried working on this issue ,Please let me know if some changes need to be made to this solution. I am open to incorporate any of your suggestions to solving this issue. Thank you!

 Changes:
https://github.com/processing/p5.js/compare/main...Garima3110:p5.js:perspective

 Screenshots of the change:

image

image

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [ ] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

File: src/webgl/p5.Camera.js [hunk 1]
```diff
@@ -127,16 +127,31 @@ p5.prototype.camera = function (...args) {
...
  * @for p5
- * @param  {Number} [fovy]   camera frustum vertical field of view,
- *                           from bottom to top of view, in <a href="#/p5/angleMode">angleMode</a> units
- * @param  {Number} [aspect] camera frustum aspect ratio
- * @param  {Number} [near]   frustum near plane length
- * @param  {Number} [far]    frustum far plane length
+ * @param  {Number} [fovy]  - camera frustum vertical field of view,
+ *                           from bottom to top of view, in <a href="#/p5/angleMode">angleMode</a> units.
+ *  The default value is now variable and based on the canvas size as:
+ *  this.defaultEyeZ = 800;
```
Comment (davepagurek) [inline src/webgl/p5.Camera.js:136]: Instead of writing this as code, maybe we can write what's going on in plain English? i.e. we pick a field of view that ensures you can see exactly `width` pixels across horizontally and `height` pixels across vertically 800 units away from the camera.

File: src/webgl/p5.Camera.js [hunk 2]
```diff
@@ -127,16 +127,31 @@ p5.prototype.camera = function (...args) {
...
+ *  defaultCameraFOV = 2 * Math.atan(this._renderer.height / 2 / this.defaultEyeZ);
+ *
+ * @param  {Number} [aspect] camera frustum aspect ratio .The default value = this._renderer.width / this._renderer.height;
+ * @param  {Number} [near]   frustum near plane length or the near clipping plane. The default value = this.defaultEyeZ * 0.1;
+ * @param  {Number} [far]    frustum far plane length or the far clipping plane. The default value = this.defaultEyeZ * 10;
+ *
+ * However,If you prefer a fixed field of view, you can use the old defaults:
```
Comment (davepagurek) [inline src/webgl/p5.Camera.js:143]: Rather than referring to these as the old defaults, we can motivate why someone might want to do this by starting with "If you want a fixed field of view". To ensure that you can see `width` across horizontally and `height` across vertically at that fixed field of view, you need to place your camera `(height/2) / tan(fovy / 2)` away from the subject, and ten go on to explain how the near/far values are derived from that.

**Review by davepagurek — COMMENTED** [review]
Thanks for taking on this task! If you run `grunt yui:dev` you can see your docs changes locally. They currently look like this:
image

image

What's going on here is that it doesn't render correctly when you place more description after the parameters. All of your description has to be above the examples and parameters sections for it to get correctly placed in the top section.

File: src/webgl/p5.Camera.js [hunk 3]
```diff
@@ -127,16 +127,31 @@ p5.prototype.camera = function (...args) {
...
+ *
+ * - Field of view: PI/3 (60 degrees)
+ * - Canvas aspect ratio: width / height
+ * - Near clipping plane: eyeZ / 10
+ * - Far clipping plane: eyeZ * 10
+ *
+ * @example
+ * Set a fixed field of view
```
Comment (davepagurek) [inline src/webgl/p5.Camera.js:151]: Was this intended to be a new example with code?

**Review by davepagurek — COMMENTED** [review]
I'm also noticing that the camera clips through the cubes in the example now. Can we maybe also adjust the `translate()` call in the example to push the scene farther back to avoid this?

**Comment by Garima3110** [comment]
> What's going on here is that it doesn't render correctly when you place more description after the parameters. All of your description has to be above the examples and parameters sections for it to get correctly placed in the top section.

I'll just look at it , and get back to you .Thanks a lot for the suggestions..! @davepagurek

File: src/webgl/p5.Camera.js (hunk 3 above)
Comment (Garima3110) [reply src/webgl/p5.Camera.js:151]: > Was this intended to be a new example with code?

yes my bad , I thought of providing a separate example for the perspective function's syntax with the old default values. But now , i think there's no need of a separate example for this. I 'll just remove this one.

**Comment by Garima3110** [comment]
@davepagurek , I have made the changes please have a look at them and let me know if some more changes need to be made. 
link to the changes:
https://github.com/processing/p5.js/commit/9d478f34874088bcaeacad63703ccaa4da6cfff9
Thankyou!

**Comment by Garima3110** [comment]
Due to some unwanted things happening in the repository I had to take a brutal step of deleting the repository . I'll get back to solving this issue very soon @davepagurek .
Thank you!
